.github/workflows/ci.yml

```diff
@@ -12,33 +12,19 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby: ['2.4', '2.5', '2.6', '2.7', '3.0', jruby-head, truffleruby-head]
-        redis: ['4']
-        search: [['opensearch-ruby:2.1.0', 'opensearchproject/opensearch:2.2.1']]
-        include:
-          # Redis 3
-          - ruby: '2.7'
-            redis: '3'
-            search: ['opensearch-ruby:2.1.0', 'opensearchproject/opensearch:2.2.1']
-          # Opensearch 1.0
-          - ruby: '2.7'
-            redis: '4'
-            search: ['opensearch-ruby:1.0.1', 'opensearchproject/opensearch:1.0.1']
-          # Elasticsearch 7.13
-          - ruby: '2.7'
-            redis: '4'
-            search: ['elasticsearch:7.13.3', 'elasticsearch:7.13.4']
-          # Redis 5
-          - ruby: '2.7'
-            redis: '5'
-            search: ['opensearch-ruby:2.1.0', 'opensearchproject/opensearch:2.2.1']
-          # Ruby 2.3 & Elasticsearch 7.5
-          - ruby: '2.3'
-            redis: '4'
-            search: ['elasticsearch:7.5.0', 'elasticsearch:7.13.4']
+        ruby: ['2.3', '2.4', '2.5', '2.6', '2.7', '3.0', '3.1', '3.2', '3.3', '3.4', head, jruby-head, truffleruby-head]
+        redis: ['3', '4', '5', '6', '7', '8']
+        search: [
+          ['opensearch-ruby:1', 'opensearchproject/opensearch:1'],
+          ['opensearch-ruby:2', 'opensearchproject/opensearch:2'],
+          ['opensearch-ruby:3', 'opensearchproject/opensearch:3'],
+          ['elasticsearch:7', 'elasticsearch:7.17.28'],
+          ['elasticsearch:8', 'elasticsearch:8.18.2'],
+          ['elasticsearch:9', 'elasticsearch:9.0.2']
+        ]
```

Comment:
This matrix is much too large to actually run. We'll want to pair it down to just a sampling of the things we want to support.

Reply:
> the things we want to support

Yes this is what I don't have context on. So we've got rubies 2.3 - 3.4 (as per the gemspec) and head versions just to catch incompatibilities early and redis gem majors 3 and 4. Would you like support the last three major versions of opensearch and elasticsearch? Or just two like redis?

Reply:
FWIW the matrix should be good now, I'm testing it in my fork: https://github.com/joshuay03/faulty/pull/1